### DISCUSSION TITLE
How ought table packages to interact with quarto?

### DISCUSSION THREAD

**Discussion by hughjonesd** [quarto-dev/quarto-cli#1068] [Q&A] [answered]
{huxtable} author here. huxtable typically spits out HTML, TeX, etc. within an `asis_output()` or `htmlPreserve()` chunk in knitr documents. This allows the package to do things that wouldn't work with markdown output, e.g. coloured backgrounds.

What's the best approach for interacting with quarto? For example, it would be nice for options like `tbl-cap` and `label` to "just work". Is there/will there be an API where I can find out these options? Or, should the package just spit out the table output, and trust quarto to tweak it?

**Comment by jjallaire** [comment, marked as answer]
`tbl-cap` will actually read the raw HTML or LaTeX table source and attempt to inject the caption and crossref label. So yes you should just spit it out and trust Quarto to tweak it (if it doesn't work then this code might not be interacting well with your HTML or LaTeX: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/filters/quarto-pre/table-captions.lua). Any problems you see should either be fixed by changes in huxtable output OR changes in our code that handles the output.

**Comment by hughjonesd** [comment]
OK, thank you. So I should just continue wrapping output in `asis_output()`. But presumably not `htmlPreserve()` since that creates a separate fragment that gets read back in. I will try this.

**Comment by jjallaire** [comment]
We actually override htmlPreserve to create a raw html block so that should work too

**Reply by hughjonesd** [reply]
I think if I use htmlPreserve, quarto labels and captions seem not to work, at least in HTML. Maybe that is a bug on your end? If so I can try to reproduce.

**Reply by jjallaire** [reply]
Yes if you could create an MRE that would be great

**Comment by hughjonesd** [comment]
This seems to work great so far. I have one more question, if I may: is there a simple way to provide table labels from within R to quarto? For example, if a user calls `label(ht) <- "my-label"` is there some way I can tell quarto "hey, @my-label should refer to this table?"

**Reply by jjallaire** [reply]
Just form a caption with the label as it would occur in a markdown caption  eg `{#tbl-foo}`

**Comment by hughjonesd** [comment]
See bug #1069.
